File: Binary Search/binarySearch.java
/*BINARY SEARCH IMPLEMENTED IN JAVA
CONDITION THE INPUT ARRAY SHOULD BE SORTED 
time complexity is O(logn)
*/
import java.util.Scanner;

public class binarySearch {
    public static void main(String[] args) {

        int[] arr = {1,2,3,4,5,6,7,8,9,10};
        int l = 0 , h = arr.length , mid ;

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number to be searched: ");
        int val = sc.nextInt();


        boolean y = true;
        while (y){
            if (l < h){
                mid = (l + h)/2;
                if (arr[mid] == val){
                    System.out.print(val+" is at index "+mid);
                    y = false;
                }else if(arr[mid] > val){
                    h = mid - 1;

                }else if(arr[mid] < val){
                    l = mid + 1 ;

                }
            }else{
                System.out.print("Element is absent");
                break;
            }
        }
        
       

        sc.close();
    }
    
}

/*
Output:

Enter the number to be searched: 10
10 is at index 9

Enter the number to be searched: 1
1 is at index 0

Enter the number to be searched: 11
Element is absent
*/

File: Binary Search/binarySearch.py
#BINARY SEARCH IMPLEMENTED IN PYTHON
#CONDITION THE INPUT ARRAY SHOULD BE SORTED 
#time complexity is O(logn)

def binary_search(arr ,s):
    low = 0
    up = len(arr)

    y = True


    while y: 
        if low < up:
            mid = low + (up - low)//2
            if arr[mid] == s:
                print("{} is at index {}".format(s,mid)) 
                y = False
            elif arr[mid] > s:
                up = mid - 1
            elif arr[mid] < s:
                low = mid + 1
        else:
            print("Element not present")
            break

            

    



a = [1,2,3,4,5,6]

s = int(input("Enter the number to be searched: "))

binary_search(a,s)
        

#Output

#Enter the number to be searched: 5
#5 is at index 4

#Enter the number to be searched: 2
#2 is at index 1

#Enter the number to be searched: 10 
#Element not present



